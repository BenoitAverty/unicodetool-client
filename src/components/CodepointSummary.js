const CodepointSummary = ({ codepoint }) => (
  <div className='CodepointSummary'>
    <p className='displayedCharacter'><strong>{codepoint.character}</strong></p>
    <div className='characterProperties'>
      <h1>{codepoint.value} {codepoint.name}</h1>
      <ul>
        <li>Block: {codepoint.properties.block}</li>
        <li>General Category: {codepoint.properties.generalCategory}</li>
      </ul>
    </div>
    <style jsx>{`
      .CodepointSummary {
        display: flex;
        align-items: center;
        background: #0275d8;
        border-radius: 0.5rem;
        color: white;
        padding: 0.5rem;
      }
      .displayedCharacter {
        flex: 0 1 auto;
        width: 80px;
        height: 80px;
        margin: 0;
        color: black;
        font-size: 60px;
        background: #eceeef;
        border-radius: 0.5rem;
        display: flex;
        justify-content: center;
        align-items: center;
      }
      .characterProperties {
        padding-left: 1rem;
      }
      .characterProperties h1 {
        font-size: 1.1rem;
        font-weight: bold;
      }
      .characterProperties ul {
        list-style-type: none;
        font-size: 0.9rem;
        padding-left: 15px;
        margin: 0;
      }
    `}</style>
  </div>
)
export default CodepointSummary
