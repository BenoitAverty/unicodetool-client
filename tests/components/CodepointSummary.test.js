import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import CodepointSummary from '../../src/components/CodepointSummary'

describe('CodepointSummary component', () => {
  describe('With a complete codepoint', () => {
    it('matches snapshot', () => {
      const codepoint = {
        value: 'U+0041',
        name: 'LATIN CAPITAL LETTER A',
        properties: {
          block: 'ASCII',
          generalCategory: 'Lu'
        }
      }

      const comp = shallow(<CodepointSummary codepoint={codepoint} />)

      expect(toJson(comp)).toMatchSnapshot()
    })
  })

  describe('With a combining mark codepoint', () => {
    it('matches snapshot', () => {
      const codepoint = {
        value: 'U+0307',
        name: 'COMBINING DOT ABOVE',
        properties: {
          block: 'Diacriticals',
          generalCategory: 'Mn'
        }
      }

      const comp = shallow(<CodepointSummary codepoint={codepoint} />)

      expect(toJson(comp)).toMatchSnapshot()
    })
  })
})
