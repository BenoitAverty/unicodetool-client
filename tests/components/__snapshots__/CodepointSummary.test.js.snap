// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodepointSummary component With a combining mark codepoint matches snapshot 1`] = `
<div
  className="CodepointSummary"
  data-jsx={769656655}
>
  <p
    className="displayedCharacter"
    data-jsx={769656655}
  >
    <strong
      data-jsx={769656655}
    >
      â—Œundefined
    </strong>
  </p>
  <div
    className="characterProperties"
    data-jsx={769656655}
  >
    <h1
      data-jsx={769656655}
    >
      U+0307
       
      COMBINING DOT ABOVE
    </h1>
    <ul
      data-jsx={769656655}
    >
      <li
        data-jsx={769656655}
      >
        Block: 
        Diacriticals
      </li>
      <li
        data-jsx={769656655}
      >
        General Category: 
        Mn
      </li>
    </ul>
  </div>
  <_class
    css=".CodepointSummary[data-jsx=\\"769656655\\"] {display:-webkit-flex; display:flex;align-items: center;background: #0275d8;border-radius: 0.5rem;color: white;padding: 0.5rem;}.displayedCharacter[data-jsx=\\"769656655\\"] {-webkit-flex: 0 1 auto;-moz-flex: 0 1 auto;flex: 0 1 auto;width: 80px;height: 80px;margin: 0;color: black;font-size: 60px;background: #eceeef;border-radius: 0.5rem;display:-webkit-flex; display:flex;justify-content: center;align-items: center;}.characterProperties[data-jsx=\\"769656655\\"] {padding-left: 1rem;}.characterProperties[data-jsx=\\"769656655\\"] h1[data-jsx=\\"769656655\\"] {font-size: 1.1rem;font-weight: bold;}.characterProperties[data-jsx=\\"769656655\\"] ul[data-jsx=\\"769656655\\"] {list-style-type: none;font-size: 0.9rem;padding-left: 15px;margin: 0;}"
    styleId={769656655}
  />
</div>
`;

exports[`CodepointSummary component With a complete codepoint matches snapshot 1`] = `
<div
  className="CodepointSummary"
  data-jsx={769656655}
>
  <p
    className="displayedCharacter"
    data-jsx={769656655}
  >
    <strong
      data-jsx={769656655}
    />
  </p>
  <div
    className="characterProperties"
    data-jsx={769656655}
  >
    <h1
      data-jsx={769656655}
    >
      U+0041
       
      LATIN CAPITAL LETTER A
    </h1>
    <ul
      data-jsx={769656655}
    >
      <li
        data-jsx={769656655}
      >
        Block: 
        ASCII
      </li>
      <li
        data-jsx={769656655}
      >
        General Category: 
        Lu
      </li>
    </ul>
  </div>
  <_class
    css=".CodepointSummary[data-jsx=\\"769656655\\"] {display:-webkit-flex; display:flex;align-items: center;background: #0275d8;border-radius: 0.5rem;color: white;padding: 0.5rem;}.displayedCharacter[data-jsx=\\"769656655\\"] {-webkit-flex: 0 1 auto;-moz-flex: 0 1 auto;flex: 0 1 auto;width: 80px;height: 80px;margin: 0;color: black;font-size: 60px;background: #eceeef;border-radius: 0.5rem;display:-webkit-flex; display:flex;justify-content: center;align-items: center;}.characterProperties[data-jsx=\\"769656655\\"] {padding-left: 1rem;}.characterProperties[data-jsx=\\"769656655\\"] h1[data-jsx=\\"769656655\\"] {font-size: 1.1rem;font-weight: bold;}.characterProperties[data-jsx=\\"769656655\\"] ul[data-jsx=\\"769656655\\"] {list-style-type: none;font-size: 0.9rem;padding-left: 15px;margin: 0;}"
    styleId={769656655}
  />
</div>
`;
